src/fetchers/tarball-fetcher.js

```diff
@@ -77,7 +77,11 @@ export default class TarballFetcher extends BaseFetcher {
   } {
     const validateStream = new crypto.HashStream();
     const extractorStream = new UnpackStream();
-    const untarStream = tar.Extract({path: this.dest, strip: 1});
+    const untarStream = tarFs.extract(this.dest, {
+      strip: 1,
+      dmode: parseInt(555, 8), // all dirs should be readable
+      fmode: parseInt(444, 8), // all files should be readable
```

Comment:
Sometimes we need to strip first folder, sometimes not.
And moving dmode,fbmode into a separate function looks like an overkill